docs/dep.md

```diff
@@ -0,0 +1,26 @@
+# kubernetes/test-infra dependency management
```

Comment:
Can we link to this from the main README so contributors can find it without digging too much?

Reply:
Done